editor/src/document/portfolio_message_handler.rs

```diff
@@ -409,15 +457,48 @@ impl MessageHandler<PortfolioMessage, &InputPreprocessorMessageHandler> for Port
 				responses.push_back(PortfolioMessage::SelectDocument { document_id: prev_id }.into());
 			}
 			RequestAboutGraphiteDialog => {
-				responses.push_back(FrontendMessage::DisplayDialogAboutGraphite.into());
+				self.about_graphite_dialog.register_properties(responses, LayoutTarget::DialogDetails);
+				responses.push_back(
+					FrontendMessage::DisplayDialog {
+						icon: "GraphiteLogo".to_string(),
+						heading: "Graphite".to_string(),
+					}
+					.into(),
+				);
+			}
+			RequestComingSoonDialog { issue } => {
+				let coming_soon = ComingSoon { issue };
+				coming_soon.register_properties(responses, LayoutTarget::DialogDetails);
+				responses.push_back(
+					FrontendMessage::DisplayDialog {
+						icon: "Warning".to_string(),
+						heading: "Coming soon".to_string(),
+					}
+					.into(),
+				);
+			}
+			RequestNewDocumentDialog => {
+				self.new_document_dialog = dialogs::NewDocument {
+					name: self.generate_new_document_name(),
+					infinite: true,
+					dimensions: glam::UVec2::new(1920, 1080),
+				};
+				self.new_document_dialog.register_properties(responses, LayoutTarget::DialogDetails);
+				responses.push_back(
+					FrontendMessage::DisplayDialog {
+						icon: "File".to_string(),
+						heading: "New document".to_string(),
+					}
+					.into(),
+				);
 			}
```

Comment:
The Portfolio is defined as, essentially, the plural collection of documents. It seems like these dialogs are more of an editor concept, not related to the group of documents. (Remember, in the future we'll be able to re-dock and drag around document tabs into other windows or close all the document tabs entirely.) To help avoid making code files so big, what do you think about defining another message handler like `dialog_message_handler.rs`? Maybe it would live in `/editor/src/dialogs` by moving the current `dialogs` folder up a level? (Only a suggestion, you have a better understanding of the structure than me.)